src/test/rpc/Feature_test.cpp

```diff
@@ -0,0 +1,274 @@
+            BEAST_EXPECTS(! feature[jss::vetoed].asBool(),
+                feature[jss::name].asString() + " vetoed");
+            BEAST_EXPECTS(feature[jss::supported].asBool(),
+                feature[jss::name].asString() + " supported");
+        }
+    }
+
+    void
+    testSingleFeature()
+    {
+        testcase ("Feature Param");
+
+        using namespace test::jtx;
+        Env env {*this};
+
+        auto jrr = env.rpc("feature", "CryptoConditions") [jss::result];
+        jrr.removeMember(jss::status);
+        BEAST_EXPECT(jrr.size() == 1);
+        auto feature = *(jrr.begin());
+
+        BEAST_EXPECTS(feature[jss::name] == "CryptoConditions", "name");
+        BEAST_EXPECTS(! feature[jss::enabled].asBool(), "enabled");
+        BEAST_EXPECTS(! feature[jss::vetoed].asBool(), "vetoed");
+        BEAST_EXPECTS(feature[jss::supported].asBool(), "supported");
+    }
+
+    void
+    testInvalidFeature()
+    {
+        testcase ("Invalid Feature");
+
+        using namespace test::jtx;
+        Env env {*this};
+
+        auto jrr = env.rpc("feature", "AllTheThings") [jss::result];
+        BEAST_EXPECT(jrr[jss::error] == "badFeature");
+        BEAST_EXPECT(jrr[jss::error_message] == "Feature unknown or invalid.");
+    }
+
+    void
+    testNonAdmin()
+    {
+        testcase ("Feature Without Admin");
+
+        using namespace test::jtx;
+        Env env {*this, []() {
+            auto p = std::make_unique<Config>();
+            test::setupConfigForUnitTests(*p);
+            (*p)["port_rpc"].set("admin","");
+            (*p)["port_ws"].set("admin","");
+            return p;
+        }()};
+
+        auto jrr = env.rpc("feature") [jss::result];
+        // The current HTTP/S ServerHandler returns an HTTP 403 error code here
+        // rather than a noPermission JSON error. The JSONRPCClient just eats that
+        // error and returns an null result.
+        BEAST_EXPECT(jrr.isNull());
+    }
+
+    void
+    testSomeEnabled()
+    {
+        testcase ("No Params, Some Enabled");
+
+        using namespace test::jtx;
+        Env env {*this,
+            features(featureSusPay),
+            features(featureCryptoConditions)};
+        // The amendment table has to be modified
+        // since that is what feature RPC actually checks
+        env.app().getAmendmentTable().enable(featureSusPay);
+        env.app().getAmendmentTable().enable(featureCryptoConditions);
+
+        auto jrr = env.rpc("feature") [jss::result];
+        if(! BEAST_EXPECT(jrr.isMember(jss::features)))
+            return;
+        for(auto it = jrr[jss::features].begin();
+            it != jrr[jss::features].end(); ++it)
+        {
+            uint256 id;
+            id.SetHexExact(it.key().asString().c_str());
+            bool shouldBeEnabled = env.app().getAmendmentTable().isEnabled(id);
+            BEAST_EXPECTS((*it)[jss::enabled].asBool() == shouldBeEnabled,
+                (*it)[jss::name].asString() + " enabled");
```

Comment:
Is it worth checking supported?
`            bool supported = env.app().getAmendmentTable().isSupported(id);
            BEAST_EXPECTS((*it)[jss::supported].asBool() == supported,
                (*it)[jss::name].asString() + " supported");`